src/infill/subDivCube.cpp

```diff
@@ -0,0 +1,279 @@
+    tilt.matrix[3] = -one_over_sqrt_6; tilt.matrix[4] = -one_over_sqrt_6; tilt.matrix[5] = sqrt_two_third ;
+    tilt.matrix[6] = one_over_sqrt_3;  tilt.matrix[7] = one_over_sqrt_3;  tilt.matrix[8] = one_over_sqrt_3;
+
+    infill_rotation_matrix = PointMatrix(infill_angle);
+    Point3Matrix infill_angle_mat(infill_rotation_matrix);
+
+    rotation_matrix = infill_angle_mat.compose(tilt);
+
+    mesh.base_subdiv_cube = new SubDivCube(mesh, center, curr_recursion_depth - 1);
+}
+
+void SubDivCube::generateSubdivisionLines(int64_t z, Polygons& result, Polygons** directional_line_groups)
+{
+    auto addLine = [&](Point from, Point to)
+    {
+        PolygonRef p = result.newPoly();
+        p.add(from);
+        p.add(to);
+    };
+    /*!
+     * Adds the defined line to the specified polygons. It assumes that the specified polygons are all parallel lines. Combines line segments with touching ends closer than epsilon.
+     * \param group the polygons to add the line to
+     * \param from the first endpoint of the line
+     * \param to the second endpoint of the line
+     */
+    auto addLineAndCombine = [&](Polygons& group, Point from, Point to)
+    {
+        int epsilon = 10;
+        for (unsigned int idx = 0; idx < group.size(); idx++)
+        {
+            if (abs(from.X - group[idx][1].X) < epsilon && abs(from.Y - group[idx][1].Y) < epsilon)
+            {
+                from = group[idx][0];
+                group.remove(idx);
+                idx--;
+                continue;
+            }
+            if (abs(to.X - group[idx][0].X) < epsilon && abs(to.Y - group[idx][0].Y) < epsilon)
+            {
+                to = group[idx][1];
+                group.remove(idx);
+                idx--;
+                continue;
+            }
+        }
+        PolygonRef p = group.newPoly();
+        p.add(from);
+        p.add(to);
+    };
+    bool top_level = false; //!< if this cube is the top level of the recursive call
+    if (directional_line_groups == nullptr) //!< if directional_line_groups is null then set the top level flag and create directional line groups
+    {
+        top_level = true;
+        directional_line_groups = (Polygons**)calloc(3, sizeof(Polygons*));
+        for (int idx = 0; idx < 3; idx++)
+        {
+            directional_line_groups[idx] = new Polygons();
+        }
+    }
+    int32_t z_diff = abs(z - center.z); //!< the difference between the cube center and the target layer.
+    if (z_diff > height[depth] / 2) //!< this cube does not touch the target layer. Early exit.
+    {
+        return;
+    }
+    if (z_diff < max_draw_z_diff[depth]) //!< this cube has lines that need to be drawn.
+    {
+        Point relative_a, relative_b; //!< relative coordinates of line endpoints around cube center
+        Point a, b; //!< absolute coordinates of line endpoints
+        relative_a.X = (square_height[depth] / 2) * ((double)(max_draw_z_diff[depth] - z_diff) / (double)max_draw_z_diff[depth]);
```

Comment:
We should keep these computations in the fixed-point `long long`s, to prevent accumulation of floating point errors. That probably means casting `square_height` and such to `long long`s instead of casting the Z coordinates to `double`s.